candlelight.go

```diff
@@ -17,27 +17,37 @@
 package candlelight
 
 import (
+	"errors"
 	"fmt"
+	"strings"
+
 	"go.opentelemetry.io/otel/exporters/stdout"
 	"go.opentelemetry.io/otel/exporters/trace/jaeger"
 	"go.opentelemetry.io/otel/exporters/trace/zipkin"
-	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
 	"go.opentelemetry.io/otel/trace"
-	"strings"
 )
 
 var (
-	nilProviderErr = fmt.Errorf("No provider is configured")
+	ErrTracerProviderNotFound    = errors.New("TracerProvider builder could not be found")
+	ErrTracerProviderBuildFailed = errors.New("Failed building TracerProvider")
 )
 
+// DefaultTracerProvider is used when no provider is given.
+// The Noop tracer provider turns all tracing related operations into
+// noops essentially disabling tracing.
+const DefaultTracerProvider = "noop"
```

Comment:
Hi @joe94 
based on the discussion in https://github.com/xmidt-org/tr1d1um/issues/203, i will suggest to use the default as stdout when the default is set as true.


i am referring to this case


```
The client explicitly asks for the default tracing behavior
This is the missing case. There should be a way for a client to request tracing but not be forced to configure all the particulars. This isn't the same as a default. It's a case where the client does want the default behavior described by the candlelight package.

The last case is usually as simple as:

tracing:
    default: true
```

Reply:
Yep, I'll include the updates to `InjectTraceInformationInLogger` and `ExtractTraceInformation` 👍

Reply:
As for the configuration setup behavior, I talked to John to get clarification on the intent behind the "default" case. The goal of that case is to help users not have to manually provide many configuration values. Since we have just a few options, we should be okay for now. 

So that leaves us with the following tracing behavior cases:
1) No configuration provided: this disables tracing. We agreed that using OTEL's noop tracer is fine in this case.
2) Configuration provided: build the tracerProvider as indicated by config.

Reply:
Sounds Great !!!

By when this PR will be merged to candlelight, Post that we will work on Tr1d1um, scytale, and other components according to these candlelight changes. so that we can release other components also.

Reply:
@Sachin4403, sounds good! Ideally, I wrap up the changes in `candlelight` and `Tr1d1um` this week so we have a model to base on work on. Then, I can write a GitHub issue for each of the services you posted on the discussions so we can team up on those 🚀   Once those issues are up, just ask to be assigned to work on it and I will assign you so we don't work on the same thing 👍